odc/stats/utils.py

```diff
@@ -132,6 +132,36 @@ def bin_annual(
     return tasks
 
 
+def bin_rolling_annual(
+    cells: dict[tuple[int, int], Cell],
+    temporal_range,
+    years: int,
+    interval: int,
+) -> dict[tuple[str, int, int], list[CompressedDataset]]:
+    binner = rolling_season_binner(
+        mk_rolling_years_rules(temporal_range, years, interval)
+    )
+
+    tasks = {}
+    for tidx, cell in cells.items():
+        # This is a great pylint warning, but doesn't apply here because we
+        # only call the lambda from inside each iteration of the loop
+        # pylint:disable=cell-var-from-loop
+        utc_offset = cell.utc_offset
+        _grouped = toolz.groupby(lambda ds: binner(ds.time + utc_offset), cell.dss)
+
+        grouped = defaultdict(list)
+        for key, value in _grouped.items():
+            for k in key:
+                grouped[k].extend(value)
+
+        for temporal_k, dss in grouped.items():
+            if temporal_k != "":
```

Comment:
It doesn't make a huge difference with simplifying the logic in this case, but if you move this if-statement to the insertion loop instead it will at least save some work/memory allocations.